### PR TITLE
Ensure there are test samples for imdb dataset, when maxlen is low

### LINKED ISSUE #40609: Test set is empty for imdb.load_data() with low maxlen value
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): custom
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: 3.6

**Describe the current behavior**
With the current imdb.load_data(), the following results are seen for different values of maxlen.

```
load_data                          (len(x_train), len(x_test))
------------------------------------------------------------ 
imdb.load_data(maxlen=50)    -->    (1035, 0)
imdb.load_data(maxlen=100)   -->    (5736, 0)
imdb.load_data(maxlen=200)   -->    (25000, 3913)
imdb.load_data()             -->    (25000, 25000)
```

<b>Analysis:</b> We can observe that when maxlen is low, the number of test samples can be 0. This is because the train and test data is concatenated, then the samples with length > maxlen are removed, and the first 25,000 are considered as training data.   

**Describe the expected behavior**

The number of test samples should not be zero.

<b>Fix:</b>This can be fixed when data can be filtered first to remove the ones with length > maxlen, and then concatenate to process further. The following are the results after the fix.

```
load_data                              (len(x_train), len(x_test))
------------------------------------------------------------ 
imdb.load_data(maxlen=50)    -->    (477, 558)
imdb.load_data(maxlen=100)   -->    (2773, 2963)
imdb.load_data(maxlen=200)   -->    (14244, 14669)
imdb.load_data()             -->    (25000, 25000)
```

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

```
from tensorflow.keras.datasets import imdb
(x_train, _), (x_test, _) = imdb.load_data(maxlen=50)
print((len(x_train), len(x_test)))  # This gives (1035, 0)

(x_train, _), (x_test, _) = imdb.load_data(maxlen=100)
print((len(x_train), len(x_test)))  # This gives (5736, 0)

(x_train, _), (x_test, _) = imdb.load_data(maxlen=200)
print((len(x_train), len(x_test)))  # This gives (25000, 3913)
```

### PR DESCRIPTION
**Pull request by dsandeep0138**
With the current imdb.load_data(), the following results are seen
for different values of maxlen.

```
	load_data                (len(x_train), len(x_test))
------------------------------------------------------------
imdb.load_data(maxlen=50)    -->    (1035, 0)
imdb.load_data(maxlen=100)   -->    (5736, 0)
imdb.load_data(maxlen=200)   -->    (25000, 3913)
imdb.load_data()             -->    (25000, 25000)
```

Analysis: We can observe that when maxlen is low, the number
of test samples can be 0. This is because the train and test data is
concatenated, then the samples with length > maxlen are removed, and
the first 25,000 are considered as training data.

Fix: This can be fixed when data can be filtered first to remove the
ones with length > maxlen, and then concatenate to process further.
The following are the results after the fix.

```
     fixed load_data              (len(x_train), len(x_test))
------------------------------------------------------------
imdb.load_data(maxlen=50)    -->    (477, 558)
imdb.load_data(maxlen=100)   -->    (2773, 2963)
imdb.load_data(maxlen=200)   -->    (14244, 14669)
imdb.load_data()             -->    (25000, 25000)
```

Fixes #40609

### CODE REVIEW

**Comment by dsandeep0138** [comment]
@googlebot I signed it!

**Review by qlzh727 — APPROVED** [review]
Thanks for the fix.
